src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.props

```diff
@@ -106,7 +109,39 @@ The .NET Foundation licenses this file to you under the MIT license.
     <Exec Command="command -v $(CppLinker)" IgnoreExitCode="true" StandardOutputImportance="Low">
       <Output TaskParameter="ExitCode" PropertyName="_WhereLinker" />
     </Exec>
+
+    <Exec Command="command -v $(CppCompilerAndLinkerAlternative)" Condition="'$(_WhereLinker)' != '0'" IgnoreExitCode="true" StandardOutputImportance="Low">
```

Comment:
This autodetection should kick-in only if the compiler/linker is not explicitly specified by the user.